tests/fsharp/Compiler/CompilerAssert.fs

```diff
@@ -179,6 +179,44 @@ let main argv = 0"""
                 try File.Delete inputFilePath with | _ -> ()
                 try File.Delete outputFilePath with | _ -> ()
 
+    let CompileOfAst source =
+        let outputFilePath = Path.ChangeExtension (Path.GetTempFileName(), ".dll")
+        let parseOptions = { FSharpParsingOptions.Default with SourceFiles = [|"test.fs"|] }
+
+        let parseResults = 
+            checker.ParseFile("test.fs", SourceText.ofString source, parseOptions) 
+            |> Async.RunSynchronously
+    
+        Assert.IsEmpty(parseResults.Errors, sprintf "Parse errors: %A" parseResults.Errors)
+        Assert.IsTrue(parseResults.ParseTree.IsSome, "no parse tree returned")
+
+        let compileErrors, statusCode = 
+            checker.Compile([parseResults.ParseTree.Value], "test", outputFilePath, []) 
```

Comment:
I notice elsewhere in this file that calls to the typechecker are being synchronized. AFAIK these api's should be threadsafe though?